ci/sync/main.sh

```diff
@@ -93,34 +93,37 @@ get_full_log() {
 }
 
 rollback_and_log() {
-    echo "ROLLBACK_BLOCK : $ROLLBACK_BLOCK"
+    echo "ROLLBACK_BLOCK : ${ROLLBACK_BLOCK}"
     ROLLBACK_HASH=$($DEFI_CLI_CMD getblockhash $((ROLLBACK_BLOCK)))
-    echo "ROLLBACK_HASH : $ROLLBACK_HASH"
+    echo "ROLLBACK_HASH : ${ROLLBACK_HASH}"
     $DEFI_CLI_CMD invalidateblock "$ROLLBACK_HASH"
     echo "Rolled back to block : $($DEFI_CLI_CMD getblockcount)"
 
     get_full_log | grep -v "AccountChange"
 }
 
 create_pre_sync_rollback_log() {
-    local DATADIR_ROLLBACK="$DATADIR-rollback"
-    local DEFID_CMD="$DEFID_BIN -datadir=$DATADIR_ROLLBACK -daemon -debug=accountchange -spv -rpcport=9999 -port=9998 -connect=0 -checkpoints=0 -interrupt-block=$((START_BLOCK + 1))"
-    local DEFI_CLI_CMD="$DEFI_CLI_BIN -datadir=$DATADIR_ROLLBACK -rpcport=9999"
-    local DEBUG_FILE="$DATADIR_ROLLBACK/debug.log"
+    local DATADIR_ROLLBACK="${DATADIR}-rollback"
+    local DEFID_CMD="${DEFID_BIN} -datadir=${DATADIR_ROLLBACK} -daemon -debug=accountchange -spv -rpcport=9999 -port=9998 -connect=0 -checkpoints=0 -interrupt-block=$((START_BLOCK + 1))"
+    local DEFI_CLI_CMD="${DEFI_CLI_BIN} -datadir=${DATADIR_ROLLBACK} -rpcport=9999"
+    local DEBUG_FILE="${DATADIR_ROLLBACK}/debug.log"
 
     cp -r "$DATADIR" "$DATADIR_ROLLBACK"
     rm -f "$DEBUG_FILE"
     $DEFID_CMD
     sleep 90
 
-    rollback_and_log >"$PRE_ROLLBACK_LOG"
+    rollback_and_log > "$PRE_ROLLBACK_LOG"
 
     $DEFI_CLI_CMD stop
+
+    # allow node to shutdown
+    sleep 10
```

Comment:
How about adding another `wait_for_shutdown` that actually checks for process to have exited?

Reply:
```
wait_for_shutdown() {
  while { 
      pgrep defid
      sleep 10
      if over TIMEOUT seconds; exit error. 
  } 
}
```

Reply:
Resolved. Refactored the main.sh script with a stop_node function, which kills the defid process (if it exists) and checks for the process exiting up to 20 seconds.